codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java

```diff
@@ -1177,13 +1160,11 @@ private String readDelimiterStandard(String delimiter) {
      *             Need more chunks and reset the readerInder to the previous
      *             value
      */
-    private String readDelimiter(String delimiter) {
-        SeekAheadOptimize sao;
-        try {
-            sao = new SeekAheadOptimize(undecodedChunk);
-        } catch (SeekAheadNoBackArrayException ignored) {
-            return readDelimiterStandard(delimiter);
+    private static String readDelimiter(ByteBuf undecodedChunk, String delimiter) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "readDelimiter" is 33 which is greater than 25 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)